barman/server.py

```diff
@@ -1896,6 +1897,12 @@ def get_required_xlog_files(
             target_tli, _, _ = xlog.decode_segment_name(end)
             calculated_target_tli = target_tli
 
+        # If --target_immediate was specified, assign the backup end_lsn value
+        # to the target_lsn so we can get the last WAL file that is required
+        # for the recovery process.
+        if target_immediate:
+            target_lsn = backup.end_xlog
+
```

Comment:
This triggers unnecessary calculations here, besides making the method a "bit more confusing", if you get what I mean.

You already have the WAL file required for consistency in the backup metadata.
I'd say to rather have an explicit `if target_immediate` in the loop.